staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -490,3 +511,58 @@ func startDBSizeMonitorPerEndpoint(client *clientv3.Client, interval time.Durati
 		cancel()
 	}, nil
 }
+
+// checkVersion checks for deprecated versions of etcd and returns an error if find any
+// currently the versions < 3.4.25, between 3.5.0 and 3.5.7
+// based on this: https://github.com/etcd-io/etcd/issues/15220
+func checkVersion(c *clientv3.Client, dialTimeOut time.Duration) (err error) {
+	eps := c.Endpoints()
+	baseCtx := context.TODO()
+
+	ctx, cancel := context.WithCancel(baseCtx)
+	if dialTimeOut > 0 {
+		cancel()
+		ctx, cancel = context.WithTimeout(baseCtx, dialTimeout)
+		defer cancel()
+	}
+	g, ctx := errgroup.WithContext(ctx)
+
+	for _, ep := range eps {
+		// if cluster is current, any endpoint gives a recent version
+		g.Go(func() error {
+			resp, rerr := c.Status(ctx, ep)
+			if rerr != nil {
+				return rerr
```

Comment:
```suggestion
				return fmt.Errorf("failed checking etcd status, endpoint: %q, err: %w", ep, rerr)
```

Reply:
I was wondering maybe we can rely on errors that created from Status, but the code is now updated